targets/TARGET_NORDIC/TARGET_MCU_NRF51822/us_ticker.c

```diff
@@ -287,6 +287,7 @@ void us_ticker_set_interrupt(timestamp_t timestamp)
 
 void us_ticker_fire_interrupt(void)
 {
+    us_ticker_callbackPending = true;
```

Comment:
This implementation should be

```
void us_ticker_fire_interrupt(void)
{
    uint32_t newCallbackTime = rtc1_getCounter() + FUZZY_RTC_TICKS;
    NRF_RTC1->CC[0]             = newCallbackTime & MAX_RTC_COUNTER_VAL;
    us_ticker_callbackTimestamp = newCallbackTime;
    us_ticker_callbackPending = true;
    rtc1_enableCompareInterrupt();
}

```

however, still fails the test case, investigating what is going on.

cc @nvlsianpu @anangl If you can spot a mistake here, let us know.

Reply:
cc @pan-

Reply:
The problem is also with NRF51_DK target, fails also for absolute zero test case.

Reply:
Tracking issue created #5284

Reply:
Note that the implementation used by the `NRF51_DK` should not be that one but rather the one in `targets/TARGET_NORDIC/TARGET_MCU_NRF51822/us_ticker.c` .

`EVTENSET` may be use to trigger the interrupt of the right channel.

Reply:
EVTENSET(the event enable set register) enables event routing, does not set the pending flag.

Reply:
NRF51 uses NRF5/us_ticker_api.c file, tested via #error 

```
targets\TARGET_NORDIC\TARGET_NRF5\us_ticker.c
        [Error] us_ticker.c@256,0:  #35: #error directive: this_one!
        [DEBUG] Return: 1
        [DEBUG] Output: "mbed-os\targets\TARGET_NORDIC\TARGET_NRF5\us_ticker.c", line 256: Error:  #35: #error directive: aaa
        [DEBUG] Output: mbed-os\targets\TARGET_NORDIC\TARGET_NRF5\us_ticker.c: 0 warnings, 1 error

[mbed] ERROR: "c:\python27\python.exe" returned error code 1.
[mbed] ERROR: Command "c:\python27\python.exe -u mbed-os\tools\test.py -t ARM -m NRF51_DK --source mbed-os --build mbed-os\BUILD\tests\NRF51_DK\ARM --test-spec mbed-os\BUILD\tests\NRF51_DK\ARM\test_spec.json -n tests-mbed_drivers-timerevent -v" in "mbed-os"
```

> EVTENSET 

how would this look like?

Reply:
That's why I pointed out that the modified file in this PR won't affect NRF51_DK because it is the wrong one: `TARGET_MCU_NRF51822/us_ticker.c`. 

@jiandeng Is correct, `EVTENSET` won't help. The main issue with the current implementation is the retrieval of the capture compare unit which has initiated the interrupt. A variable which store the state of CC unit IRQ triggered by software may be sufficient to solve current the issue. Of course the interrupt handler code have to be updated as well.